tests/integration/fuzz_transaction/mod.rs

```diff
@@ -687,17 +687,27 @@ async fn multiple_connections_fuzz(opts: FuzzOptions) {
 
                 match operation {
                     Operation::Begin { concurrent } => {
-                        shared_shadow_db.begin_transaction(next_tx_id, false);
-                        if concurrent {
-                            // in tursodb, BEGIN CONCURRENT immediately starts a transaction.
-                            shared_shadow_db.take_snapshot_if_not_exists(next_tx_id);
-                        }
-                        *current_tx_id = Some(next_tx_id);
-                        next_tx_id += 1;
-
                         let query = operation.to_string();
 
-                        conn.execute(query.as_str(), ()).await.unwrap();
+                        let result = conn.execute(query.as_str(), ()).await;
```

Comment:
This is just something I noticed while fuzzing this branch using a higher `weight_begin_concurrent` -- since `BEGIN CONCURRENT` can fail, we cannot start the shadow transaction until we know it succeeded in the real db.